libgit2-sys/lib.rs

```diff
@@ -3854,6 +3855,82 @@ extern "C" {
     ) -> c_int;
 }
 
+#[repr(C)]
+pub struct git_merge_file_options {
+    pub version: c_uint,
+
+    /// Label for the ancestor file side of the conflict which will be prepended
+    /// to labels in diff3-format merge files.
+    pub ancestor_label: *const c_char,
+
+    /// Label for our file side of the conflict which will be prepended
+    /// to labels in merge files.
+    pub our_label: *const c_char,
+
+    /// Label for their file side of the conflict which will be prepended
+    /// to labels in merge files.
+    pub their_label: *const c_char,
+
+    /// The file to favor in region conflicts.
+    pub favor: git_merge_file_favor_t,
+
+    /// see `git_merge_file_flag_t`
+    pub flags: c_uint,
+    pub marker_size: c_ushort,
+}
+
+#[repr(C)]
+#[derive(Copy, Clone)]
+pub struct git_merge_file_input {
+    pub version: c_uint,
+    /// Pointer to the contents of the file.
+    pub ptr: *const c_char,
+    /// Size of the contents pointed to in `ptr`.
+    pub size: size_t,
+    /// File name of the conflicted file, or `NULL` to not merge the path.
+    pub path: *const c_char,
+    /// File mode of the conflicted file, or `0` to not merge the mode.
+    pub mode: c_uint,
+}
+
+#[repr(C)]
+#[derive(Copy, Clone)]
+pub struct git_merge_file_result {
+    /// True if the output was automerged, false if the output contains
+    /// conflict markers.
+    pub automergeable: c_uint,
+
+    /// The path that the resultant merge file should use, or NULL if a
+    /// filename conflict would occur.
+    pub path: *const c_char,
+
+    /// The mode that the resultant merge file should use.
+    pub mode: c_uint,
+
+    /// The contents of the merge.
+    pub ptr: *const c_char,
+
+    /// The length of the merge contents.
+    pub len: size_t,
+}
+
+extern "C" {
```

Comment:
Could this get merged (hah!) with the general block of function definitions elsewhere in this file?

Reply:
I was worrying to make it mess at the beginning so they were put together, now they've been moved to the right places.